File: src/.NET/fix50/ConfirmationAck.cs
namespace QuickFix50
{

  public class ConfirmationAck : Message
  {
    public ConfirmationAck() : base(MsgType()) {}
    static QuickFix.MsgType MsgType() { return new QuickFix.MsgType("AU"); }

    public ConfirmationAck(
      QuickFix.ConfirmID aConfirmID,
      QuickFix.TradeDate aTradeDate,
      QuickFix.TransactTime aTransactTime,
      QuickFix.AffirmStatus aAffirmStatus )
    : base(MsgType()) {
      set(aConfirmID);
      set(aTradeDate);
      set(aTransactTime);
      set(aAffirmStatus);
    }

    public void set(QuickFix.ConfirmID value)
    { setField(value); }
    public QuickFix.ConfirmID get(QuickFix.ConfirmID  value)
    { getField(value); return value; }
    public QuickFix.ConfirmID getConfirmID()
    { QuickFix.ConfirmID value = new QuickFix.ConfirmID();
      getField(value); return value; }
    public bool isSet(QuickFix.ConfirmID field)
    { return isSetField(field); }
    public bool isSetConfirmID()
    { return isSetField(664); }

    public void set(QuickFix.TradeDate value)
    { setField(value); }
    public QuickFix.TradeDate get(QuickFix.TradeDate  value)
    { getField(value); return value; }
    public QuickFix.TradeDate getTradeDate()
    { QuickFix.TradeDate value = new QuickFix.TradeDate();
      getField(value); return value; }
    public bool isSet(QuickFix.TradeDate field)
    { return isSetField(field); }
    public bool isSetTradeDate()
    { return isSetField(75); }

    public void set(QuickFix.TransactTime value)
    { setField(value); }
    public QuickFix.TransactTime get(QuickFix.TransactTime  value)
    { getField(value); return value; }
    public QuickFix.TransactTime getTransactTime()
    { QuickFix.TransactTime value = new QuickFix.TransactTime();
      getField(value); return value; }
    public bool isSet(QuickFix.TransactTime field)
    { return isSetField(field); }
    public bool isSetTransactTime()
    { return isSetField(60); }

    public void set(QuickFix.AffirmStatus value)
    { setField(value); }
    public QuickFix.AffirmStatus get(QuickFix.AffirmStatus  value)
    { getField(value); return value; }
    public QuickFix.AffirmStatus getAffirmStatus()
    { QuickFix.AffirmStatus value = new QuickFix.AffirmStatus();
      getField(value); return value; }
    public bool isSet(QuickFix.AffirmStatus field)
    { return isSetField(field); }
    public bool isSetAffirmStatus()
    { return isSetField(940); }

    public void set(QuickFix.ConfirmRejReason value)
    { setField(value); }
    public QuickFix.ConfirmRejReason get(QuickFix.ConfirmRejReason  value)
    { getField(value); return value; }
    public QuickFix.ConfirmRejReason getConfirmRejReason()
    { QuickFix.ConfirmRejReason value = new QuickFix.ConfirmRejReason();
      getField(value); return value; }
    public bool isSet(QuickFix.ConfirmRejReason field)
    { return isSetField(field); }
    public bool isSetConfirmRejReason()
    { return isSetField(774); }

    public void set(QuickFix.MatchStatus value)
    { setField(value); }
    public QuickFix.MatchStatus get(QuickFix.MatchStatus  value)
    { getField(value); return value; }
    public QuickFix.MatchStatus getMatchStatus()
    { QuickFix.MatchStatus value = new QuickFix.MatchStatus();
      getField(value); return value; }
    public bool isSet(QuickFix.MatchStatus field)
    { return isSetField(field); }
    public bool isSetMatchStatus()
    { return isSetField(573); }

    public void set(QuickFix.Text value)
    { setField(value); }
    public QuickFix.Text get(QuickFix.Text  value)
    { getField(value); return value; }
    public QuickFix.Text getText()
    { QuickFix.Text value = new QuickFix.Text();
      getField(value); return value; }
    public bool isSet(QuickFix.Text field)
    { return isSetField(field); }
    public bool isSetText()
    { return isSetField(58); }

    public void set(QuickFix.EncodedTextLen value)
    { setField(value); }
    public QuickFix.EncodedTextLen get(QuickFix.EncodedTextLen  value)
    { getField(value); return value; }
    public QuickFix.EncodedTextLen getEncodedTextLen()
    { QuickFix.EncodedTextLen value = new QuickFix.EncodedTextLen();
      getField(value); return value; }
    public bool isSet(QuickFix.EncodedTextLen field)
    { return isSetField(field); }
    public bool isSetEncodedTextLen()
    { return isSetField(354); }

    public void set(QuickFix.EncodedText value)
    { setField(value); }
    public QuickFix.EncodedText get(QuickFix.EncodedText  value)
    { getField(value); return value; }
    public QuickFix.EncodedText getEncodedText()
    { QuickFix.EncodedText value = new QuickFix.EncodedText();
      getField(value); return value; }
    public bool isSet(QuickFix.EncodedText field)
    { return isSetField(field); }
    public bool isSetEncodedText()
    { return isSetField(355); }

  };

}

File: src/java/src/quickfix/fix44/ConfirmationAck.java
package quickfix.fix44;
import quickfix.FieldNotFound;
import quickfix.Group;
import quickfix.field.*;

public class ConfirmationAck extends Message
{
  public ConfirmationAck()
  {
    getHeader().setField(new MsgType("AU"));
  }
  public ConfirmationAck(
    quickfix.field.ConfirmID aConfirmID,
    quickfix.field.TradeDate aTradeDate,
    quickfix.field.TransactTime aTransactTime,
    quickfix.field.AffirmStatus aAffirmStatus ) {

    getHeader().setField(new MsgType("AU"));
    set(aConfirmID);
    set(aTradeDate);
    set(aTransactTime);
    set(aAffirmStatus);
  }

  public void set(quickfix.field.ConfirmID value)
  { setField(value); }
  public quickfix.field.ConfirmID get(quickfix.field.ConfirmID  value)
    throws FieldNotFound
  { getField(value); return value; }
  public quickfix.field.ConfirmID getConfirmID() throws FieldNotFound
  { quickfix.field.ConfirmID value = new quickfix.field.ConfirmID();
    getField(value); return value; }
  public boolean isSet(quickfix.field.ConfirmID field)
  { return isSetField(field); }
  public boolean isSetConfirmID()
  { return isSetField(664); }
  public void set(quickfix.field.TradeDate value)
  { setField(value); }
  public quickfix.field.TradeDate get(quickfix.field.TradeDate  value)
    throws FieldNotFound
  { getField(value); return value; }
  public quickfix.field.TradeDate getTradeDate() throws FieldNotFound
  { quickfix.field.TradeDate value = new quickfix.field.TradeDate();
    getField(value); return value; }
  public boolean isSet(quickfix.field.TradeDate field)
  { return isSetField(field); }
  public boolean isSetTradeDate()
  { return isSetField(75); }
  public void set(quickfix.field.TransactTime value)
  { setField(value); }
  public quickfix.field.TransactTime get(quickfix.field.TransactTime  value)
    throws FieldNotFound
  { getField(value); return value; }
  public quickfix.field.TransactTime getTransactTime() throws FieldNotFound
  { quickfix.field.TransactTime value = new quickfix.field.TransactTime();
    getField(value); return value; }
  public boolean isSet(quickfix.field.TransactTime field)
  { return isSetField(field); }
  public boolean isSetTransactTime()
  { return isSetField(60); }
  public void set(quickfix.field.AffirmStatus value)
  { setField(value); }
  public quickfix.field.AffirmStatus get(quickfix.field.AffirmStatus  value)
    throws FieldNotFound
  { getField(value); return value; }
  public quickfix.field.AffirmStatus getAffirmStatus() throws FieldNotFound
  { quickfix.field.AffirmStatus value = new quickfix.field.AffirmStatus();
    getField(value); return value; }
  public boolean isSet(quickfix.field.AffirmStatus field)
  { return isSetField(field); }
  public boolean isSetAffirmStatus()
  { return isSetField(940); }
  public void set(quickfix.field.ConfirmRejReason value)
  { setField(value); }
  public quickfix.field.ConfirmRejReason get(quickfix.field.ConfirmRejReason  value)
    throws FieldNotFound
  { getField(value); return value; }
  public quickfix.field.ConfirmRejReason getConfirmRejReason() throws FieldNotFound
  { quickfix.field.ConfirmRejReason value = new quickfix.field.ConfirmRejReason();
    getField(value); return value; }
  public boolean isSet(quickfix.field.ConfirmRejReason field)
  { return isSetField(field); }
  public boolean isSetConfirmRejReason()
  { return isSetField(774); }
  public void set(quickfix.field.MatchStatus value)
  { setField(value); }
  public quickfix.field.MatchStatus get(quickfix.field.MatchStatus  value)
    throws FieldNotFound
  { getField(value); return value; }
  public quickfix.field.MatchStatus getMatchStatus() throws FieldNotFound
  { quickfix.field.MatchStatus value = new quickfix.field.MatchStatus();
    getField(value); return value; }
  public boolean isSet(quickfix.field.MatchStatus field)
  { return isSetField(field); }
  public boolean isSetMatchStatus()
  { return isSetField(573); }
  public void set(quickfix.field.Text value)
  { setField(value); }
  public quickfix.field.Text get(quickfix.field.Text  value)
    throws FieldNotFound
  { getField(value); return value; }
  public quickfix.field.Text getText() throws FieldNotFound
  { quickfix.field.Text value = new quickfix.field.Text();
    getField(value); return value; }
  public boolean isSet(quickfix.field.Text field)
  { return isSetField(field); }
  public boolean isSetText()
  { return isSetField(58); }
  public void set(quickfix.field.EncodedTextLen value)
  { setField(value); }
  public quickfix.field.EncodedTextLen get(quickfix.field.EncodedTextLen  value)
    throws FieldNotFound
  { getField(value); return value; }
  public quickfix.field.EncodedTextLen getEncodedTextLen() throws FieldNotFound
  { quickfix.field.EncodedTextLen value = new quickfix.field.EncodedTextLen();
    getField(value); return value; }
  public boolean isSet(quickfix.field.EncodedTextLen field)
  { return isSetField(field); }
  public boolean isSetEncodedTextLen()
  { return isSetField(354); }
  public void set(quickfix.field.EncodedText value)
  { setField(value); }
  public quickfix.field.EncodedText get(quickfix.field.EncodedText  value)
    throws FieldNotFound
  { getField(value); return value; }
  public quickfix.field.EncodedText getEncodedText() throws FieldNotFound
  { quickfix.field.EncodedText value = new quickfix.field.EncodedText();
    getField(value); return value; }
  public boolean isSet(quickfix.field.EncodedText field)
  { return isSetField(field); }
  public boolean isSetEncodedText()
  { return isSetField(355); }
}

